docs/provider/bitwarden-secrets-manager.md

```diff
@@ -0,0 +1,136 @@
+    --set bitwarden-sdk-server.enabled=true
+```
+
+##### Certificate
+
+The Bitwarden SDK Server _NEEDS_ to run as an HTTPS service. That means that any installation that once to with Bitwarden
+provider will need to generate a certificate. The best approach for that is to use cert-manager. It's easy to set up
+and can generate a certificate that the store can use to connect with the server.
+
+For a sample set up look at the bitwarden sdk server's test setup. It contains a self-signed certificate issuer for
+cert-manager.
+
+### External secret store
+
+With that out of the way, let's take a look at how a secret store would look like.
+
+```yaml
+{% include 'bitwarden-secrets-manager-secret-store.yaml' %}
+```
+
+The api url and identity url are optional. The secret should contain the token for the Machine account for bitwarden.
+
+!!! note inline end
+Make sure that the machine account has Read-Write access to the Project that the secrets are in.
+
+!!! note inline end
+A secret store is organization/project dependent. Meaning a 1 store == 1 organization/project. This is so that we ensure
+that no other project's secrets can be modified accidentally _or_ intentionally.
+
+### External Secrets
+
+There are two ways to fetch secrets from the provider.
+
+#### Find by UUID
+
+In order to fetch a secret by using its UUID simply provide that as remote key in the external secrets like this:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: bitwarden
+spec:
+  refreshInterval: 10s
+  secretStoreRef:
+    # This name must match the metadata.name in the `SecretStore`
+    name: bitwarden-secretsmanager
+    kind: SecretStore
+  data:
+  - secretKey: test
+    remoteRef:
+      key: "339062b8-a5a1-4303-bf1d-b1920146a622"
+```
+
+#### Find by Name
+
+To find a secret using its name, we need a bit more information. Mainly, these are the rules to find a secret:
+
+- if name is a UUID get the secret
+- if name is NOT a UUID Property is mandatory that defines the projectID to look for
+- if name + projectId + organizationId matches, we return that secret
+- if more than one name exists for the same projectId within the same organization we error
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: bitwarden
+spec:
+  refreshInterval: 10s
+  secretStoreRef:
+    # This name must match the metadata.name in the `SecretStore`
+    name: bitwarden-secretsmanager
+    kind: SecretStore
+  data:
+  - secretKey: test
+    remoteRef:
+      key: "secret-name"
+```
+
+### Push Secret
+
+Pushing a secret is also implemented. Pushing a secret requires even more restrictions because Bitwarden Secrets Manager
+allows creating the same secret with the same key multiple times. In order to avoid overwriting, or potentially, returning
+the wrong secret, we restrict push secret with the following rules:
+
+- property == projectID which is mandatory field for PushSecret
```

Comment:
I think this can be removed as the projectID is now in the SecretStore.

Reply:
Thank you! Will do. :) <3 <3 <3